src/main/java/org/traccar/protocol/XirgoProtocolDecoder.java

```diff
@@ -233,10 +233,18 @@ private Object decodeCustom(
                     dateBuilder.setSecond(Integer.parseInt(time[2]));
                     break;
                 case "LT":
-                    position.setLatitude(Double.parseDouble(values[i]));
+                    if (values[i].contains(".")) {
+                        position.setLatitude(Double.parseDouble(values[i]));
+                    } else {
+                        position.setLatitude(Double.parseDouble(values[i]) / 1000000);
```

Comment:
This can probably be optimized by using `parseInt` instead. Same for other places.